### PR TITLE
[Core] Rename `PromptInputs` and `inputs`, with backwards compatibility

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR re-implements #8673, with backwards compatibility for:

- Make the old name `PromptInputs` still available
- `inputs` keyword argument in various methods of `LLMEngine`, `AsyncLLMEngine` and `vllm.engine.multiprocessing`.

FIX https://github.com/vllm-project/vllm/pull/8673#issuecomment-2369755880

### CODE REVIEW

File: vllm/engine/multiprocessing/client.py [hunk 1]
```diff
@@ -399,17 +430,51 @@ def generate(
...
+    @overload
+    def encode(
+        self,
+        prompt: PromptType,
+        pooling_params: PoolingParams,
+        request_id: str,
+        lora_request: Optional[LoRARequest] = None,
+        trace_headers: Optional[Mapping[str, str]] = None,
+    ) -> AsyncGenerator[EmbeddingRequestOutput, None]:
+        ...
```
Comment (njhill) [inline vllm/engine/multiprocessing/client.py:463]: Is this overload really needed (and equivalent for the other methods)? Is it just the fact that `prompt` arg is Optional?

That could just be a comment instead? I.e. that the arg won't be optional in future..

**Review by njhill — COMMENTED** [review]
Thanks @DarkLight1337 looks good, but per my other comment I was thinking we should probably keep both `PromptInputs` and `PromptType` for now, before removing after the next release?

File: vllm/engine/multiprocessing/client.py (hunk 1 above)
Comment (DarkLight1337) [reply vllm/engine/multiprocessing/client.py:463]: They are to clearly specify the two versions of the method. `prompt` arg isn't actually optional, it is only like that in the implementation in order to be compatible with both overloads.

**Comment by DarkLight1337** [comment]
> Thanks @DarkLight1337 looks good, but per my other comment I was thinking we should probably keep both `PromptInputs` and `PromptType` for now, before removing after the next release?

I have added this, PTAL.

File: vllm/inputs/data.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ class TokensPrompt(TypedDict):
 the user desires to express both the encoder & decoder
 prompts explicitly, i.e. :class:`ExplicitEncoderDecoderPrompt`
 
-A prompt of type :class:`SingletonPromptInputs` may be employed
+A prompt of type :class:`SingletonPromptType` may be employed
```
Comment (njhill) [inline vllm/inputs/data.py:49]:
```suggestion
A prompt of type :class:`SingletonPrompt` may be employed
```

File: vllm/inputs/data.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ class ExplicitEncoderDecoderPrompt(TypedDict, Generic[_T1_co, _T2_co]):
 
     The encoder and decoder prompts, respectively,
     may formatted according to any of the
-    :class:`SingletonPromptInputs` schemas, and are not
+    :class:`SingletonPromptType` schemas, and are not
```
Comment (njhill) [inline vllm/inputs/data.py:75]:
```suggestion
    :class:`SingletonPrompt` schemas, and are not
```

File: vllm/inputs/data.py [hunk 4]
```diff
@@ -81,15 +81,15 @@ class ExplicitEncoderDecoderPrompt(TypedDict, Generic[_T1_co, _T2_co]):
     be used as an input to a decoder-only model,
     and that the `encoder_prompt` and `decoder_prompt`
     fields of this data structure themselves must be
-    :class:`SingletonPromptInputs` instances.
+    :class:`SingletonPromptType` instances.
```
Comment (njhill) [inline vllm/inputs/data.py:84]:
```suggestion
    :class:`SingletonPrompt` instances.
```

**Comment by DarkLight1337** [comment]
Updated.

**Comment by njhill** [comment]
> > Thanks @DarkLight1337 looks good, but per my other comment I was thinking we should probably keep both `PromptInputs` and `PromptType` for now, before removing after the next release?
> 
> I have added this, PTAL.

Thanks @DarkLight1337 ... my main concern was integrating code that might import/reference that type, but I guess that's not too likely anyhow and it would likely only be the concrete types like `LLMInputs` used.

**Review by njhill — APPROVED** [review]

**Pull request merged**